### PR TITLE
UXG-1246: adds agents settings

### PR DESCRIPTION
**Pull request by nicmarson**
http://jira.hq.idx.org/browse/UXG-1246

### CODE REVIEW

File: src/vue/backend/src/templates/AgentsSettings.vue [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    --space-4: 16px;
+    --space-5: 20px;
+    --space-6: 24px;
+    --font-size-label: 16px;
+    --font-size-p: 16px;
+    --line-height-label: 22px;
+    --line-height-p: 28px;
+    font-size: var(--font-size-p);
+    line-height: var(--line-height-p);
+    input[type=color], input[type=date], input[type=datetime-local], input[type=datetime], input[type=email], input[type=month], input[type=number], input[type=password], input[type=search], input[type=tel], input[type=text], input[type=time], input[type=url], input[type=week], select, textarea {
```
Comment (sgreenl5) [inline src/vue/backend/src/templates/AgentsSettings.vue:100]: Why do you have so many specifications on this? We are not using an input other than text

File: src/vue/backend/src/store/modules/agentSettings/index.js [hunk 2]
```diff
@@ -3,6 +3,11 @@ import getters from './getters'
 import mutations from './mutations'
 
 const state = {
+    deregisterMainCss: false,
+    numberOfPosts: '',
+    directorySlug: '',
```
Comment (sgreenl5) [inline src/vue/backend/src/store/modules/agentSettings/index.js:8]: Missing default 'employees'

File: src/vue/backend/src/store/modules/agentSettings/index.js [hunk 3]
```diff
@@ -3,6 +3,11 @@ import getters from './getters'
 import mutations from './mutations'
 
 const state = {
+    deregisterMainCss: false,
+    numberOfPosts: '',
```
Comment (sgreenl5) [inline src/vue/backend/src/store/modules/agentSettings/index.js:7]: Missing default 9

File: src/vue/backend/src/templates/AgentsSettings.vue [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+                ></idx-toggle-slider>
+            </idx-block>
+        </idx-form-group>
+        <idx-form-group>
+            <idx-form-label customClass="control-label" for="website-wrapper"><strong>Default Number of Posts</strong> Default Number of Posts Description of the Default Number of Posts setting in IMPress Agents. Lorem ipsum dolor sit amet, consectetur adipiscing elit.</idx-form-label>
+            <idx-form-input
+                type="text"
+                id="website-wrapper"
+                :value="numberOfPosts"
+                @change="agentSettingsStateChange({ key: numberOfPosts, value: $event.target.value })"
```
Comment (sgreenl5) [inline src/vue/backend/src/templates/AgentsSettings.vue:22]: Key needs to be a kebab cased string, here and at lines 31, 43, and 52

**Review by sgreenl5 — CHANGES_REQUESTED** [review]
The state is not currently being set properly, as all keys are undefined. Additionally, the style updates to `listingsGeneral.vue` were not included

File: src/vue/backend/src/templates/AgentsSettings.vue (hunk 1 above)
Comment (nicmarson) [reply src/vue/backend/src/templates/AgentsSettings.vue:100]: this is to override some styles at the Wordpress level

**Review by sgreenl5 — APPROVED** [review]

**Review by ralls — APPROVED** [review]
👍

**Pull request merged**